### PR TITLE
Implement level tagging

### PR DESCRIPTION
**Pull request by Beyley**
This is the first part of #260, implementing in-game level tagging for LBP1.

This also exposes an APIv3 endpoint for getting a list of levels by tag, and also shows the tags of a level on API level responses.

### CODE REVIEW

File: Refresh.GameServer/Database/GameDatabaseProvider.cs [hunk 1]
```diff
@@ -70,6 +70,7 @@ protected GameDatabaseProvider(IDateTimeProvider time)
         typeof(GameReview),
         typeof(DisallowedUser),
         typeof(RateReviewRelation),
+        typeof(TagLevelRelation),
```
Comment (jvyden) [inline Refresh.GameServer/Database/GameDatabaseProvider.cs:73]: no db schema bump?

File: Refresh.GameServer/Types/Levels/Tags.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
+using System.Text;
+
+namespace Refresh.GameServer.Types.Levels;
+
+public enum Tags
```
Comment (jvyden) [inline Refresh.GameServer/Types/Levels/Tags.cs:5]: This needs to be renamed to `Tags`; enums should never be plural. Especially when they're not flags.

Could also be a `byte`.

**Review by jvyden — CHANGES_REQUESTED** [review]

File: Refresh.GameServer/Database/GameDatabaseProvider.cs (hunk 1 above)
Comment (Beyley) [reply Refresh.GameServer/Database/GameDatabaseProvider.cs:73]: realm doesn't need it for just adding new types, it only matters when an existing type changes or gets removed(?)

File: Refresh.GameServer/Types/Levels/Tags.cs (hunk 2 above)
Comment (Beyley) [reply Refresh.GameServer/Types/Levels/Tags.cs:5]: i couldnt point a finger on why the name felt so wrong to say/type, will change to `Tag`, and make it a `byte`

**Review by jvyden — APPROVED** [review]

**Pull request merged**